### PR TITLE
Feature/signin and signup procedures

### PR DESCRIPTION
**Pull request by kikkomep**
This PR introduces several updates to the Docker image:

* switch to a new base image (the old one is no longer available);
* add the possibility to choose repo and branch of the metadata backend;
* update the entrypoint;
* add an optional utility script which allows to wait for MySQL service to become available

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,48 @@
-FROM comicrelief/php7-slim:7.1
+FROM crs4/php7-base:7.1
```
Comment (pcm32) [inline Dockerfile:1]: Excellent! much happier that we depend on our own base if not depending on an official one!

File: Dockerfile [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+# Install backend dependencies
+RUN php composer.phar install && chmod a+x setup.sh
+
+# Set environment
+ENV MYSQL_HOST "localhost"
+ENV MYSQL_PORT 3306
+ENV SERVICE_PATH "/var/www/html/portal-metadata-backend"
+ENV PATH ${SERVICE_PATH}/vendor/propel/propel/bin:${PATH}
+
+# Add utility to wait for MySQL service
```
Comment (pcm32) [inline Dockerfile:40]: This waiting logic really should be at the container orchestrator level (which I think we have), but having it here additionally doesn't hurt. But for our deployment, I would like the db waiting logic to be handled by Kubernetes through the init container.

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,48 @@
-FROM comicrelief/php7-slim:7.1
+FROM crs4/php7-base:7.1
 
 LABEL Description="Metadata backend for the PhenoMeNal Portal"
 LABEL software="PhenoMeNal Portal"
 LABEL software.version="1.1.2"
 LABEL version="0.2.2"
```
Comment (pcm32) [inline Dockerfile]: Given that you are expected to use the container through the `entrypoint.sh`, I would deem that as an "API change", which would mean that you should bump the major version number in the `version` label. Please do so as part of this PR.

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# set web server port
+web_server_port=${1:-8080}
+
+# update configuration files and initialise DB
+"${SERVICE_PATH}/setup.sh"
+
+# start the web server
+"${SERVICE_PATH}/start.sh" ${web_server_port}
```
Comment (pcm32) [inline entrypoint.sh:10]: This whole process seems simpler and with less commands than the longer starting line that we had in the helm chart for portal. Good! Are all these commands available in setup.sh and start.sh or we really never needed all that complication? I wonder...

**Review by pcm32 — CHANGES_REQUESTED** [review]
Excellent work @kikkomep. My only request would be the bumping of the major version number in the version Label. Thanks!

File: Dockerfile (hunk 3 above)
Comment (kikkomep) [reply Dockerfile]: Ok! Major version has been updated via https://github.com/phnmnl/container-portal-metadata-backend/pull/1/commits/1a904da3d5dc3ec2255436233cd47f292c6185d0

**Review by pcm32 — APPROVED** [review]
Thanks @kikkomep!

File: Dockerfile (hunk 2 above)
Comment (kikkomep) [reply Dockerfile:40]: Yes, @pcm32! The waiting logic is mainly intended to instantiate the container without Kubernetes (e.g., with Docker Compose, etc.) and it is optional. Thus for our deployment we can use the db waiting logic handled by Kubernetes through the init container.

File: entrypoint.sh (hunk 4 above)
Comment (kikkomep) [reply entrypoint.sh:10]: The long list of commands of the old starting line has been reorganized in a more consistent way on the two setup.sh and start.sh scripts: the former contains all the logic to configure the MySQL connection and to initialize the MySQL DB; the latter simply starts the PHP daemon for serving the App.

**Pull request merged**